paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -105,10 +106,23 @@ public void createDatabase(String name, boolean ignoreIfExists)
             }
             throw new DatabaseAlreadyExistException(name);
         }
+        createDatabaseImpl(name, properties);
+    }
 
-        createDatabaseImpl(name);
+    @Override
+    public Map<String, String> loadDatabaseProperties(String name)
+            throws DatabaseNotExistException {
+        if (isSystemDatabase(name)) {
+            return Collections.emptyMap();
+        }
+        if (!databaseExists(name)) {
+            throw new DatabaseNotExistException(name);
+        }
+        return loadDatabasePropertiesImpl(name);
     }
 
+    protected abstract Map<String, String> loadDatabasePropertiesImpl(String name);
```

Comment:
I prefer to name this method `loadDatabaseProperties0` if this is required.

Reply:
Just for unified code style, like createDatabase